R/joss_article.R

```diff
@@ -0,0 +1,91 @@
+#' Journal of Open Source Software (JOSS) format.
+#'
+#' Format for creating a Journal of Open Source Software (JOSS) or Journal of
+#' Open Source Education (JOSE) articles. Adapted
+#' from \url{https://github.com/openjournals/whedon}. As these journals take
+#' articles as markdown, this format can be used to generate markdown from
+#' R Markdown and to locally preview how the article will appear as PDF.
+#'
+#' @inheritParams rmarkdown::pdf_document
+#' @param journal one of "JOSS" or"JOSE"
+#' @param keep_md Whether to retain the intermediate markdown and images.
+#'   Defaults to TRUE.
+#' @param doi,year,volume,issue,page,submitted,published,review_url,repository,archive_doi
+#'   Article metadata, all optional for generating PDF preview, auto-generated
+#'   on submission.
+#' @param latex_engine,... Arguments passed to \code{rmarkdown::pdf_document}
+#' @export
+joss_article <- function(journal = "JOSS",
+                         keep_md = TRUE,
+                         doi = "",
+                         year = "",
+                         volume = "",
+                         issue = "",
+                         page = "",
+                         submitted = "",
+                         published = "",
+                         review_url = "",
+                         repository = "",
+                         archive_doi = "",
+                         latex_engine = "xelatex",
+                         ...) {
+
+  rmarkdown::pandoc_available('2.2', TRUE)
+
+  logo_path <- find_resource("joss_article",
+                             paste0(journal, "-logo.png"))
+  journalname <- ifelse(journal == "JOSS",
+                        "Journal of Open Source Software",
+                        "Journal of Open Source Education")
+
+  template_variables <- c(
+      paste0("logo_path=", logo_path),
+      paste0("year=", year),
+      paste0("journal_name=", journalname),
+      paste0("formatted_doi=", doi),
+      paste0("archive_doi=https://doi.org/", archive_doi),
+      paste0("review_issue_url=", review_url),
+      paste0("repository=", repository),
+      paste0("submitted=", submitted),
+      paste0("published=", published),
+      paste0("issue=", issue),
+      paste0("volume=", volume),
+      paste0("page=", page),
+      "graphics=true"
+    )
+
+  template_variables <- c(rbind(rep("-V", length(template_variables)),
+                                template_variables))
+
+  pandoc_args <- c(
+    template_variables,
+    "--csl", find_resource("joss_article", "apa.csl")
```

Comment:
I hope the `csl` option could be set in the YAML metadata of the Rmd document, too.

Reply:
I wanted to avoid doing this because I wanted the format to compile correctly on a standard `paper.md` without modifying the YAML.

Reply:
All other templates in this package include `csl` under the `skeleton/` directory (instead of `resources/`) so the csl file will be copied over when creating a new article using `rticles::draft()`, e.g. https://github.com/rstudio/rticles/tree/master/inst/rmarkdown/templates/acm_article/skeleton Then in Rmd: https://github.com/rstudio/rticles/blob/master/inst/rmarkdown/templates/acm_article/skeleton/skeleton.Rmd#L15

I'm aware of the potential disadvantages of this approach (e.g. the `csl` file may change in the upstream), but I'd like this template to be consistent with other templates. Similarly, I hope `logo_path`, `journal_name`, and `graphics` variables could all be set in the YAML frontmatter of `skeleton.Rmd` (the png logos need to be moved to `skeleton/` accordingly).